### ISSUE TITLE
switch language too easily in the same sentence

### ISSUE THREAD

**Issue by adrienlafond** [Kieirra/murmure#438]
### Murmure version

1.10.1

### Operating system

macOS (Apple Silicon)

### Display server (Linux only)

None

### Feature impacted

Transcription

### Bug description

switch language too easily in the same sentence, Which end up being a mix of French and English. 

### Steps to reproduce

Difficult to reproduce myself, but this has been something very annoying recently. 
It happens only than when I speak French 
then some English words are added into it or sometimes it's a translation of the French words I have said. 

### Debug logs

```shell

```

**Comment by Kieirra** [comment]
Hi, and thank you for creating the issue.

Murmure runs a fine-tuned Parakeet, an ASR model that detects the language from the audio signal. There is no way to force a language, and the fine-tuning was done exactly to reduce these switches to English. So it should not happen often anymore. When it does, the cause is almost always the audio itself, a laptop built-in mic, background noise or a low input volume.

Two things worth checking:

1. Record yourself with the same mic in another app (QuickTime, Audacity) and listen back. If the audio is noisy, muffled or quiet, that is your answer.
2. Check the input volume is at 100% in the macOS sound settings.

Also worth updating to 1.11.3. It pads each chunk with silence before inference and adapts the voice detection to the mic gain, both help here.

Full guide: https://docs.murmure.app/troubleshooting/transcription/#murmure-transcribes-in-the-wrong-language

**Comment by thiswillbeyourgithub** [comment]
Hi @Kieirra ,

I'm ever closer to publishing much improved and compact models of parakeet-v3. With a general one as well as a medical finetune.

Do you think it would be relevant to murmure for me to finetune language specific variants? I hve all the tooling ready to make the model favor french, without losing too much other languages.

**Comment by thiswillbeyourgithub** [comment]
Update: I feel relatively confident that there are ways to adress this **if** you'd be willing to add a general UI setting for "which languages does the user commonly speak". I could then use pre computed language vectors and bias the model. There are many ways to go about this but maybe some could work with minimal loss and performance hit. I have ideas to make this work for all transcripts without user specifying the lang.

**Comment by Kieirra** [comment]
> add a general UI setting for "which languages does the user commonly speak"

I'm open to it. The issue is that I don't want to make the binary heavier. So either the language bias should use a file that adds little to the disk footprint, or the differences between the fine-tuned variants should be significant enough to justify a Models tab where users can switch between models, perhaps with an onboarding process.

Otherwise, if you have a fine-tuned model that is better in French and does not sacrifice the other EU languages, I'm happy to test it.

**Comment by thiswillbeyourgithub** [comment]
Alternatively, I could fine tune on some language and forget others. For example portugese is supposed to be supported but actually has high wer to begin with I think. You'd have to tell me which languages you wan to support.

**Comment by Kieirra** [comment]
Here are the languages I would keep.

I took the WER numbers from the vanilla parakeet-tdt-0.6b-v3 model https://huggingface.co/nvidia/parakeet-tdt-0.6b-v3 and drew the line at 20%. 

Must keep: Italian 3.0, Spanish 3.5, Portuguese 4.8, English 4.9, German 5.0, French 5.2, Russian 5.5, Ukrainian 6.8, Polish 7.3, Dutch 7.5, Slovak 8.8, Czech 11.0, Romanian 12.4, Croatian 12.5, Bulgarian 12.6, Finnish 13.2, Swedish 15.1, Hungarian 15.7, Estonian 17.7, Danish 18.4.

Can be dropped: Slovenian 24.0, Latvian 22.8, Greek 20.7, Maltese 20.5, Lithuanian 20.4.

> Alternatively, I could fine tune on some language and forget others. For example portugese is supposed to be supported but actually has high wer to begin with I think.

Portuguese seems to be good

**Comment by thiswillbeyourgithub** [comment]
Well 20% means one every 5 words, right? So it does not sound good enough to be used.

Keeping 5% would mean:
Italian 3.0, Spanish 3.5, Portuguese 4.8, English 4.9, German 5.0

And 10%:
French 5.2, Russian 5.5, Ukrainian 6.8, Polish 7.3, Dutch 7.5, Slovak 8.8

So 11 kept, 14 dropped.

Thoughts? Wouldn't CER be a better metric actually?

**Comment by luc-marongiu** [comment]
> Well 20% means one every 5 words, right ? So it does not sound good enough to be used.

Yep, that's my reasoning.

> So 11 kept, 14 dropped.

I made the 20% cutoff because I received feedback from a Greek user saying that it was not really usable in their language. Below, I'm not sure that's still the case, which is why I'm being a bit conservative here.

> Thoughts ? Wouldn't CER be a better metric actually ?

They did not expose CER in the link, and I guess it would be kind of the same thing, especially on a "clean" dataset like FLEURS.
